db/setup.sql

```diff
@@ -0,0 +1,108 @@
+CREATE TABLE IF NOT EXISTS users (
+  id uuid not null primary key,
+  password varchar(64) not null, -- bcrypt stores hashed password together with its salt.
+  first_name varchar(64) not null,
+  last_name varchar(64) not null,
+  email varchar(64) not null unique,
+  cash_position bigint default 1000000
+);
+
+CREATE INDEX ON users (first_name);
+CREATE INDEX ON users (last_name);
+CREATE INDEX ON users (email);
+CREATE INDEX ON users (cash_position);
+
+CREATE TABLE IF NOT EXISTS transactions (
+  id uuid NOT NULL PRIMARY KEY,
+  user_id uuid NOT NULL references users(id),
+  ticker_symbol varchar(8) NOT NULL,
+  exchange varchar(8) NOT NULL,
+  transaction_type smallint NOT NULL,
+  amount int NOT NULL,
+  strike_price int NOT NULL,
+  time_entered timestamp NOT NULL
+);
+
+comment on column transactions.transaction_type is 'Enum: buy: 0, sell: 1';
+comment on column transactions.strike_price is 'The price in US cents that the whole trade was agreed at.';
+comment on column transactions.amount is 'The number of shares traded.';
+
+CREATE INDEX ON transactions (user_id);
+CREATE INDEX ON transactions (strike_price);
+CREATE INDEX ON transactions (ticker_symbol, exchange);
+
+CREATE TABLE IF NOT EXISTS positions (
+  id uuid NOT NULL PRIMARY KEY,
+  user_id uuid NOT NULL references users(id),
+  ticker_symbol varchar(8) NOT NULL,
+  exchange varchar(8) NOT NULL,
+  amount int NOT NULL
+);
+
+CREATE INDEX ON positions (user_id);
+CREATE INDEX ON positions (ticker_symbol, exchange);
+
+CREATE TABLE IF NOT EXISTS friendships (
+  watching_user uuid PRIMARY KEY references users(id),
+  watched_user uuid references users(id)
+);
+
+
+CREATE TABLE IF NOT EXISTS watchlist (
+  id uuid NOT NULL PRIMARY KEY,
+  user_id uuid NOT NULL references users(id),
+  ticker_symbol varchar(8) NOT NULL,
+  exchange varchar(8) NOT NULL
+);
+
+CREATE INDEX ON watchlist (user_id);
+CREATE INDEX ON watchlist (ticker_symbol, exchange);
+
+-- Sample values.
+
+INSERT INTO users
```

Comment:
I would say separate out these mock insert queries into a separate file for production, but it's fine for our purposes in MVP.